test/Npgsql.Tests/Types/EnumTests.cs

```diff
@@ -384,6 +387,26 @@ await using (var __ = await GetTempTableName(conn, out var table))
             }
         }
 
+        [Test]
+        public async Task WriteUnmappedEnum()
+        {
+            await using var conn = await OpenConnectionAsync();
+            await using var _ = DeferAsync(() => conn.ExecuteNonQueryAsync("DROP TYPE IF EXISTS mood_unmapped"));
+            await conn.ExecuteNonQueryAsync($"CREATE TYPE mood_unmapped AS ENUM ('sad', 'ok', 'happy')");
```

Comment:
Check out GetTempTypeName (for example in UnmappedEnum above, which should probably be renamed to ReadUnmappedEnum). GetTempTypeName makes sure you get a unique type name (so no risk of conflict across tests), and gives you a disposable that drops the type for you.

Reply:
Not in that case since the enum has a name specified by `PgNameAttribute`.

Reply:
Right, makes sense.